File: Elitesportweb/models/Member.js
import mongoose from 'mongoose'

const MemberSchema = new mongoose.Schema({
  memberId: { 
    type: String, 
    unique: true, 
    default: () => 'ESA' + Date.now().toString().slice(-8) + Math.floor(Math.random() * 100).toString().padStart(2, '0')
  },
  fullName: { type: String, required: true },
  email: { type: String, required: true, unique: true },
  phone: { type: String, required: true },
  nic: { type: String, required: true },
  address: { type: String, required: true },
  dateOfBirth: { type: Date, required: true },
  gender: { type: String, required: true, enum: ['male', 'female'] },
  weight: { type: Number, required: true },
  height: { type: Number, required: true },
  emergencyContact: { type: String, required: true },
  medicalConditions: { type: String },
  profilePicture: { type: String },
  membershipType: { type: String, required: true },
  assignedClasses: [{ type: String }],
  joinDate: { type: Date, default: Date.now },
  status: { type: String, default: 'active' }
})

// Clear any existing model to avoid conflicts
if (mongoose.models.Member) {
  delete mongoose.models.Member
}

export default mongoose.model('Member', MemberSchema)

File: Elitesportweb/elite_sports_flutter/lib/models/member.dart
class Member {
  final String id;
  final String memberId;
  final String fullName;
  final String email;
  final String phone;
  final DateTime joinDate;
  final DateTime dateOfBirth;
  final String gender;
  final String address;
  final String emergencyContact;
  final String? medicalConditions;
  final String? profilePicture;

  Member({
    required this.id,
    required this.memberId,
    required this.fullName,
    required this.email,
    required this.phone,
    required this.joinDate,
    required this.dateOfBirth,
    required this.gender,
    required this.address,
    required this.emergencyContact,
    this.medicalConditions,
    this.profilePicture,
  });

  factory Member.fromJson(Map<String, dynamic> json) {
    return Member(
      id: json['_id'] ?? '',
      memberId: json['memberId'] ?? '',
      fullName: json['fullName'] ?? '',
      email: json['email'] ?? '',
      phone: json['phone'] ?? '',
      joinDate: DateTime.tryParse(json['joinDate'] ?? '') ?? DateTime.now(),
      dateOfBirth: DateTime.tryParse(json['dateOfBirth'] ?? '') ?? DateTime.now(),
      gender: json['gender'] ?? '',
      address: json['address'] ?? '',
      emergencyContact: json['emergencyContact'] ?? '',
      medicalConditions: json['medicalConditions'],
      profilePicture: json['profilePicture'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'memberId': memberId,
      'fullName': fullName,
      'email': email,
      'phone': phone,
      'joinDate': joinDate.toIso8601String(),
      'dateOfBirth': dateOfBirth.toIso8601String(),
      'gender': gender,
      'address': address,
      'emergencyContact': emergencyContact,
      'medicalConditions': medicalConditions,
      'profilePicture': profilePicture,
    };
  }
}